includes/purge-set-object.batch.inc

```diff
@@ -0,0 +1,129 @@
+ *   Batch Set ID.
+ *
+ * @param array $context
+ *   Batch context.
+ */
+function islandora_batch_batch_purge_object_in_set_operation($set_id, &$context) {
+  module_load_include('inc', 'islandora_batch', 'includes/ingest.batch');
+  module_load_include('inc', 'islandora_batch', 'includes/db');
+  $sandbox =& $context['sandbox'];
+  if (!islandora_batch_get_lock($set_id)) {
+    // Failed to acquire the lock... Try again in another iteration...
+    $context['finished'] = 0;
+    return;
+  }
+
+  if (!isset($sandbox['total'])) {
+    $sandbox['items'] = islandora_batch_get_queued_items_for_set($set_id);
+    $sandbox['total'] = count($sandbox['items']);
+    $sandbox['progress'] = 0;
+    if ($sandbox['total'] === 0) {
+      $context['message'] = t('Nothing to process...');
+      return;
+    }
+  }
+  // Might be increased to avoid bootstrapping overhead.
+  $slice_size = 30;
+  $slice_stop = $sandbox['progress'] + $slice_size;
+
+  for (; $sandbox['progress'] < $slice_stop && $sandbox['progress'] < $sandbox['total']; $sandbox['progress']++) {
+    $purge = FALSE;
+    $item_id = $sandbox['items'][$sandbox['progress']];
+    $object = islandora_object_load($item_id);
+    if ($object) {
+      if ($object->repository->purgeObject($object->id)) {
+        $purge = TRUE;
+      }
+    }
+    if ($purge) {
+      $context['results']['purged'][] = $item_id;
+    }
+    else {
+      $context['results']['purged_fail'][] = $item_id;
+    }
+  }
+  $context['finished'] = $sandbox['progress'] / $sandbox['total'];
+}
+
+/**
+ * Finished function for deleting ingested objects.
+ *
+ * @param array $success
+ *   An array of success passed from the batch.
+ * @param array $results
+ *   An array of results passed from the batch.
+ * @param array $operations
+ *   An array of operations passed from the batch.
+ */
+function islandora_batch_batch_purge_object_in_set_finished($success, $results, $operations) {
+  if (isset($results['purged'])) {
+    $purged_count = count($results['purged']);
+    drupal_set_message(t('Purged @count @label.', array(
+      '@count' => $purged_count,
+      '@label' => format_plural($purged_count, 'object', 'objects'),
+    )), 'warning');
+  }
+  else {
+    drupal_set_message('No objects were purged', 'warning');
+  }
+  if (isset($results['purged_fail'])) {
+    $file_uri = drupal_tempnam('public://', 'batch-set-object-purge-fail-');
+    $file = new stdClass();
+    $file->uri = $file_uri;
+    $file->filename = drupal_basename($file_uri);
+    $file->filemime = 'application/octet-stream';
+    $file->status = 0;
+    file_put_contents($file_uri, implode("\n", $results['purged_fail']));
+    file_save($file);
+
+    $failed_count = count($results['purged_fail']);
+    drupal_set_message(t('Failed to purge @count @label. Download report !link.', array(
+      '@count' => $failed_count,
+      '@label' => format_plural($failed_count, 'object', 'objects'),
+      '!link' => l(t('here'), file_create_url($file_uri)),
```

Comment:
Coder will complain, due to the unfiltered text. Any markup should generally be made explicit, like:
```php
$replacements = array();
if ($failed_count > 1) {
  module_load_include('inc', 'islandora', 'includes/utilities');
  $file = file_save_data(implode("\n", $results['purged_fail']), 'public://batch-set-object-purge-fail.txt'));
  $file = islandora_temp_file_entry($file->uri, 'text/plain');
  $replacements['@report_url'] = file_create_url($file->uri);
}
else {
  $replacements['@first_pid'] = reset($results['purged_fail']);
} 
drupal_set_message(format_plural($failed_count, 'Failed to purge 1 object: @first_pid', 'Failed to purge @count objects. Download report <a href="@report_url">here</a>.', $replacements));
```